python/tskit/tables.py

```diff
@@ -589,6 +589,23 @@ def append(self, row):
             }
         )
 
+    def replace_with(self, other):
+        # Overwrite the contents of this table with a copy of the other table
+        params = {}
+        for column in self.column_names:
+            try:
+                params[column] = getattr(other, column)
+            except AttributeError:
+                raise TypeError(
+                    "Replacement table has wrong type: it lacks a {column} column"
+                )
+        try:
+            # Not all tables have a metadata_schema: if they do, encode it with repr
+            params["metadata_schema"] = repr(other.metadata_schema)
```

Comment:
Why use ``repr`` here and not a direct API? Seem indirect and brittle (we might change repr to something slightly different, perhaps)

Reply:
That's fine internally maybe, but it's not a great convention. A well named method would be much clearer to the reader